Datastore/src/Operation.php

```diff
@@ -614,12 +750,40 @@ public function commit(array $mutations, array $options = [])
             'databaseId' => $this->databaseId,
         ];
 
-        $res = $this->connection->commit($options + [
-            'mode' => ($options['transaction']) ? 'TRANSACTIONAL' : 'NON_TRANSACTIONAL',
+        foreach ($mutations as &$mutation) {
+            $mutationType = array_keys($mutation)[0];
+            $data = $mutation[$mutationType];
+            if (isset($data['properties'])) {
+                foreach ($data['properties'] as &$property) {
+                    list($type, $val) = $this->toGrpcValue($property);
+
+                    $property[$type] = $val;
+                }
+            }
+
+            $mutation[$mutationType] = $data;
+
+            $mutation = $this->serializer->decodeMessage(new Mutation(), $mutation);
+        }
+
+        $options += [
+            'mode' => ($options['transaction']) ? Mode::TRANSACTIONAL : Mode::NON_TRANSACTIONAL,
             'mutations' => $mutations,
             'projectId' => $this->projectId,
-        ]);
+        ];
 
+        if (is_null($options['transaction'])) {
```

Comment:
Was this check there before? Or did we need it specially for v2?

Reply:
Required especially for V2 as V1 surfaces check for optional arguments before setting them [[code pointer](https://github.com/googleapis/google-cloud-php/blob/9ad2e3d7fd8358ef7f9d8ca9c6c4db68dbcadb78/Datastore/src/V1/Gapic/DatastoreGapicClient.php#L377-L379)]